### PR TITLE
MGMT-2464 - update state machine and instruction for day2 OCP

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
/assign @danielerez 
/assign @ori-amizur

### CODE REVIEW

File: internal/host/host.go [hunk 1]
```diff
@@ -571,9 +571,14 @@ func (m *Manager) selectRole(ctx context.Context, h *models.Host, db *gorm.DB) (
 		log              = logutil.FromContext(ctx, m.log)
 	)
 
-	if swag.StringValue(h.Kind) == models.HostKindAddToExistingClusterHost {
+	if isDay2Host(h) {
 		return autoSelectedRole, nil
 	}
+	/*
```
Comment (ori-amizur) [inline internal/host/host.go:577]: Please remove

File: internal/host/validator.go [hunk 2]
```diff
@@ -448,9 +448,14 @@ func (v *validator) isAPIVipConnected(c *validationContext) validationStatus {
 	if c.inventory == nil {
 		return ValidationPending
 	}
-	if swag.StringValue(c.host.Kind) != models.HostKindAddToExistingClusterHost {
+	if !isDay2Host(c.host) {
 		return ValidationSuccess
 	}
+	/*
```
Comment (ori-amizur) [inline internal/host/validator.go:454]: Remove

File: internal/host/validator.go [hunk 3]
```diff
@@ -472,9 +477,14 @@ func (v *validator) printAPIVipConnected(c *validationContext, status validation
 }
 
 func (v *validator) belongsToMajorityGroup(c *validationContext) validationStatus {
-	if swag.StringValue(c.host.Kind) == models.HostKindAddToExistingClusterHost {
+	if isDay2Host(c.host) {
 		return ValidationSuccess
 	}
+	/*
```
Comment (ori-amizur) [inline internal/host/validator.go:483]: Remove

File: internal/host/validator.go [hunk 4]
```diff
@@ -490,9 +500,14 @@ func (v *validator) belongsToMajorityGroup(c *validationContext) validationStatu
 func (v *validator) printBelongsToMajorityGroup(c *validationContext, status validationStatus) string {
 	switch status {
 	case ValidationSuccess:
-		if swag.StringValue(c.host.Kind) == models.HostKindAddToExistingClusterHost {
+		if isDay2Host(c.host) {
 			return "Day2 host is not required to be connected to other hosts in the cluster"
 		}
+		/*
```
Comment (ori-amizur) [inline internal/host/validator.go:506]: Remove

File: internal/bminventory/inventory.go [hunk 5]
```diff
@@ -1198,15 +1198,17 @@ func (b *bareMetalInventory) refreshClusterHosts(ctx context.Context, cluster *c
...
-		addHostsSkipStatuses := []string{
-			models.HostStatusAddedToExistingCluster,
-			models.HostStatusInstallingInProgress,
-			models.HostStatusInstalling,
-		}
-		if swag.StringValue(host.Kind) == models.HostKindAddToExistingClusterHost &&
-			funk.ContainsString(addHostsSkipStatuses, swag.StringValue(host.Status)) {
-			continue
-		}
+		/*
```
[the remark this answers is not in the document]
Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go:1201]: fixed

File: internal/bminventory/inventory.go [hunk 6]
```diff
@@ -1705,8 +1707,11 @@ func (b *bareMetalInventory) RegisterHost(ctx context.Context, params installer.
 
 	url := installer.GetHostURL{ClusterID: params.ClusterID, HostID: *params.NewHostParams.HostID}
 	kind := swag.String(models.HostKindHost)
-	if swag.StringValue(cluster.Kind) == models.ClusterKindAddHostsCluster {
+	switch {
```
[the remark this answers is not in the document]
Comment (yevgeny-shnaidman) [reply internal/bminventory/inventory.go:1710]: fixed

File: internal/host/host.go (hunk 1 above)
Comment (yevgeny-shnaidman) [reply internal/host/host.go:577]: fixed

File: internal/host/validator.go (hunk 2 above)
Comment (yevgeny-shnaidman) [reply internal/host/validator.go:454]: fixed

File: internal/host/validator.go (hunk 3 above)
Comment (yevgeny-shnaidman) [reply internal/host/validator.go:483]: fixed

File: internal/host/validator.go (hunk 4 above)
Comment (yevgeny-shnaidman) [reply internal/host/validator.go:506]: fixed

File: internal/host/instructionmanager.go [hunk 7]
```diff
@@ -112,9 +112,14 @@ func (i *InstructionManager) GetNextSteps(ctx context.Context, host *models.Host
 
 	returnSteps := models.Steps{}
 	stateToSteps := i.installingClusterStateToSteps
-	if swag.StringValue(host.Kind) == models.HostKindAddToExistingClusterHost {
+	if isDay2Host(host) {
 		stateToSteps = i.addHostsClusterToSteps
 	}
+	/*
```
Comment (ori-amizur) [inline internal/host/instructionmanager.go:118]: Remove

Comment (yevgeny-shnaidman) [reply internal/host/instructionmanager.go]: fixed

File: internal/host/transition.go [hunk 8]
```diff
@@ -521,12 +512,13 @@ func (th *transitionHandler) PostRefreshHost(reason string) stateswitch.PostTran
...
 
-func (th *transitionHandler) IsAddToExistingClusterHost(sw stateswitch.StateSwitch, args stateswitch.TransitionArgs) (bool, error) {
+func (th *transitionHandler) IsDay2Host(sw stateswitch.StateSwitch, args stateswitch.TransitionArgs) (bool, error) {
 	sHost, ok := sw.(*stateHost)
 	if !ok {
 		return false, errors.New("HasClusterError incompatible type of StateSwitch")
 	}
-	return swag.StringValue(sHost.host.Kind) == models.HostKindAddToExistingClusterHost, nil
+	return isDay2Host(sHost.host), nil
+	//return swag.StringValue(sHost.host.Kind) == models.HostKindAddToExistingClusterHost, nil
```
Comment (ori-amizur) [inline internal/host/transition.go:521]: Remove

File: internal/host/statemachine.go [hunk 9]
```diff
@@ -276,7 +276,8 @@ func NewHostStateMachine(th *transitionHandler) stateswitch.StateMachine {
 			stateswitch.State(models.HostStatusInstallingInProgress),
 			stateswitch.State(models.HostStatusInstalled),
 		},
-		Condition:        th.HasClusterError,
+		//Condition:        th.HasClusterError,
```
Comment (ori-amizur) [inline internal/host/statemachine.go:279]: Remove

File: internal/host/transition.go (hunk 8 above)
Comment (yevgeny-shnaidman) [reply internal/host/transition.go]: fixed

File: internal/host/statemachine.go (hunk 9 above)
Comment (yevgeny-shnaidman) [reply internal/host/statemachine.go]: fixed

**Review by ori-amizur — APPROVED** [review]
/lgtm
/approve

**Pull request merged**